src/Compiler/Compiler.cpp

```diff
@@ -712,6 +723,41 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
   };
   interp->registerCmd("synth_options", synth_options, this, 0);
 
+  auto add_litex_ip_catalog = [](void* clientData, Tcl_Interp* interp, int argc,
```

Comment:
after few month of development this file might have few thousands line of code. it could be moved out of here